### ISSUE TITLE
Increase range of extraction to shorter wavelength

### ISSUE THREAD

**Issue by julienguy** [desihub/desispec#1009]
We have been using for commissioning so far a conservative wavelength range, starting extractions at 3600A when we could go to shorter wavelength. If we want to change this, we would better do it now than later.
A suggestion is to start extractions 3576A (integer number of 0.8A steps to 3600A, so that we don't have to redefine the whole grid) and close to the shortest wavelength among all 5000 fibers that is more than 10 pixels away from the CCD edge (=3575.4A  for fiber 249 of SP5 (=SM9)). The choice of 10 pixels is a bit arbitrary. Arguments are that we need a bit of margin for calibration and possible future moves in the optics.

We would need however to demonstrate we can calibrate the data down to this wavelength.

![minwave](https://user-images.githubusercontent.com/5192160/97060508-628a6e00-1548-11eb-9ebb-c6659360673e.png)

There is a bit of signal at these short wavelength. Here is for example the average spectrum of a bright star in tile 67230.

![mean-spectrum-fiber-2780-tile-67230](https://user-images.githubusercontent.com/5192160/97060640-d75da800-1548-11eb-8570-47ba2f78f725.png)

**Comment by moustakas** [comment]
To play devil's advocate for a moment, what if we went even shorter, to say ~3520 or 3530 A? There's still the question of whether we can calibrate that blue, but for at least some fraction of the fibers we would have useful data to those wavelengths. We would then need to make sure the inverse variance spectra of the other fibers are padded with zeros, but presumably that's the default.

The main argument in favor of going bluer where we can would be: *why not?*

**Comment by djschlegel** [comment]
People will definitely use as blue as you go.  But for any given exposure+camera, the blue limit will be set by where you have at least one sky fiber and one standard star going that blue.

Lots of people hated that in SDSS I dynamically chose the blue limit for each exposure.  To avoid that hate mail, choose a blue limit (and a red limit!) at the extremes of what we'll ever get and use those everywhere.

**Comment by sbailey** [comment]
Agreed with David that whatever limits we use, they should be consistent across all exposures and targets (though some targets could start with a bunch of ivar=0 bins).

Two notes:
  * when extending this, let's also address #1006 to make it possible to algorithmically determine the default extraction ranges for each camera, instead of having that buried within `desi_proc` (e.g. so that sims can use the same range without having to inspect a real data output file or otherwise hardcode the same ranges)
  * regarding John's question "why not?" -- an example would be if the very low S/N regime introduced biases that were not reflected by the increased error bars and started causing other problems; we'd want to fix those problems simultaneous with extending the range, but it might not be as easy as lowering the limit and being done.

**Comment by julienguy** [comment]
So, well noted, we will work on this possibility to extract data at much shorter wavelength. This will require code development because it is assumed in many places in the code for now that all fibers see all wavelength. It's not a top priority for now so we probably won't have this at the beginning of recommissioning.
